uefi-sct/Doc/UserGuide/SecureBootTestKeyGenerationAndSetupGuide.md

```diff
@@ -0,0 +1,85 @@
+# Secure Boot Test Key Generation Guide - AARCH64 (Linux)
+
+Secure Boot is a key security mechanism that ensures a system boots only with firmware, operating systems, and drivers that are verified and trusted by the Original Equipment Manufacturer (OEM). It is designed to prevent the execution of unauthorized or malicious code during the boot process.
+
+## Core Features of Secure Boot
+
+- **Code Signing:** Uses public/private key cryptography.
+- **Certificate Validation:** Verifies firmware and software components using certificates.
+- **Policy Enforcement:** Prevents execution of unsigned or tampered code.
+- **Key Hierarchy:**
+  - **PK (Platform Key):** Top-level trust anchor; authorizes changes to KEK.
+  - **KEK (Key Exchange Key):** Manages updates to DB and DBX.
+  - **DB (Database of Allowed Signatures):** Stores trusted certificates and hashes.
+  - **DBX (Forbidden Signature Database):** Stores blacklisted certificates and hashes.
+
+## Supported Secure Boot Test Case
+
+### 1. `ImageLoadingTest`
+- Validates image authentication and signature mechanisms during UEFI boot.
+- Ensures that only images signed per platform policy are allowed to load.
+
+### 2. `VariableAttributes`
+- Validates the attributes (e.g., non-volatile, authenticated write access) of critical Secure Boot variables: PK, KEK, db, dbx.
+
+### 3. `VariableUpdates`
+- Tests runtime update paths for Secure Boot variables to ensure UEFI-enforced access control policies are in place.
+
+---
+
+## Pre-requisites
+
+### 1. Install Required Tools
+
+Run the following command to install the necessary tools:
+
```

Comment:
Please, mention which tool is needed before describing how to install it on specific operating systems.